app/models/metrics/firefox_new_tab_metrics_factory.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from typing import List, Dict
+from asyncio import gather
+
+import aioboto3
+
+from app import config
+from aws_xray_sdk.core import xray_recorder
+
+from app.models.metrics.firefox_new_tab_metrics_model import FirefoxNewTabMetricsModel
+from app.models.metrics.metrics_model import MetricsModel
+
+
+# Unfortunately boto3 doesn't use types for feature store records at the moment, so we define one ourselves.
+# For example records see tests/assets/json/firefox_new_tab_engagement.json
+FeatureStoreRecordType = List[Dict[str, str]]
+
+
+class FirefoxNewTabMetricsFactory():
```

Comment:
Apropos of your request for input in the video, I think this name and the other name are both fine for now. Eventually we may want to name these for their sources (Dynamo and FeatureGroups) if we end up using them—and especially this one—to populate additional surfaces as the app grows, but that's not a thing we need to do right now.